airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -67,6 +68,12 @@ def run(self, parsed_args: argparse.Namespace) -> Iterable[str]:
         if not cmd:
             raise Exception("No command passed")
 
+        if parsed_args.debug:
+            self.logger.setLevel(logging.DEBUG)
```

Comment:
note: This does actually override the `TRACE` debug level, but going through the code, I don't see it used anywhere and it appears to be legacy code from the old custom built AirbyteDebugger that was removed in favor of pythons default logger. Also occasionally introduces bugs:
https://airbytehq-team.slack.com/archives/C02TL38U5L7/p1649731849925669

Given the lack of usage, I'm partial yanking the entire TRACE message from the logger entirely

Reply:
should we check the log level with `self.logger.getEffectiveLevel()` and fail if it's `TRACE` (and add a comment for future reference)

Reply:
I think the issue is that by default we used to always set the logger to `TRACE` which has a value of 5. So it was inclusive of all logs higher than it. So at this moment, it will always be `TRACE` and it would fail.

Something I just thought of is changing the TRACE log to `15`. And if `--debug` is not present, then set level to TRACE. That way debug is exclusive by default, but we still capture TRACE at all levels. A little hacky, but TRACE is already a hack haha

Reply:
I'm like 99% confident we can just remove the `TRACE` level (as long as it's not backwards breaking) -- do you know if that's an easy removal?

Reply:
Yep should be straightforward in the logger, it just the reference to the leveling and we'll use the remaining presets